pkg/sandbox/infra.go

```diff
@@ -0,0 +1,70 @@
+package sandbox
+
+import (
+	"fmt"
+
+	"github.com/cri-o/cri-o/internal/storage"
+	libconfig "github.com/cri-o/cri-o/pkg/config"
+	"github.com/cri-o/cri-o/pkg/container"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/opencontainers/runtime-tools/generate"
+)
+
+func (s *sandbox) InitInfraContainer(serverConfig *libconfig.Config, podContainer *storage.ContainerInfo) error {
+	var err error
+	s.infra, err = container.New()
+	if err != nil {
+		return err
+	}
+
+	// determine pause command before initializing generator
+	// so a failure will not result in a half configured generator
+	pauseCommand, err := PauseCommand(serverConfig, podContainer.Config)
+	if err != nil {
+		return err
+	}
+
+	g := s.infra.Spec()
+	g.HostSpecific = true
+	g.ClearProcessRlimits()
+
+	// setup defaults for the pod sandbox
+	g.SetRootReadonly(true)
+
+	// configure default ulimits
+	for _, u := range serverConfig.Ulimits() {
+		g.AddProcessRlimits(u.Name, u.Hard, u.Soft)
+	}
+	g.SetProcessArgs(pauseCommand)
+
+	return nil
+}
+
+// Must be called after a successful call to InitInfraContainer
```

Comment:
"Can only be called only ..."?

Reply:
Also, a docstring should start with the method name, as in
```go
// Spec <does this and that>...
```

It's merely a convention, but a good one. Not sure why our linters do not catch this -- afaik they should

Reply:
Possibly this is because it is an internal package.